pkg/operator/k8sutil/configmap.go

```diff
@@ -62,11 +62,22 @@ func GetOperatorSetting(context context.Context, clientset kubernetes.Interface,
 }
 
 func GetValue(data map[string]string, settingName, defaultValue string) string {
-	if settingValue, ok := data[settingName]; ok {
-		logger.Infof("%s=%q (configmap)", settingName, settingValue)
-		return settingValue
-	} else if settingValue, ok := os.LookupEnv(settingName); ok {
-		logger.Infof("%s=%q (env var)", settingName, settingValue)
+	var settingValue, settingSource string
+
+	if val, ok := data[settingName]; ok {
+		settingValue = val
+		settingSource = "configmap"
+	} else if val, ok := os.LookupEnv(settingName); ok {
+		settingValue = val
+		settingSource = "env var"
+	}
+
+	if settingValue != "" {
```

Comment:
We discussed a different approach in huddle. These settings are already repetitive in the log since they rarely change. With a package variable (perhaps a `map[string]string`) to remember the settings, we only need to log them once each time the operator evaluates them, and then only log them again if they change.

Reply:
The reason I used this is to minimize the repetitive code, if you notice I have to have the same block of code for the env source and configmap source, that's why I used it to have a more generalised or less repetitive code

Reply:
This line makes sense, I was just documenting our discussion in case others were interested